File: python/51-100/70. Climbing Stairs.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n < 0 :
            return 0
        if n == 1:
            return 1
        slow = 1
        fast = 1
        for i in range(2, n + 1):
            temp = fast
            fast = slow + fast
            slow = temp
        return fast
            

File: cpp/51-100/70. Climbing Stairs.cpp
class Solution {
public:
	int climbStairs(int n) {
		if(n == 0)
			return 0;
		if(n == 1)
			return 1;
		if(n == 2)
			return 2;
		int pre = 1, cur = 2;
		for(int i = 3; i <= n; i++)
		{
			int temp = cur;
			cur = cur + pre;
			pre = temp;
		}
		return cur;
	}
};